settings/ChangePassword/ChangePassword.js

```diff
@@ -0,0 +1,223 @@
+        values={fullUserName}
+      />
+    );
+
+    this.callout.sendCallout({ message: successMessage });
+  };
+
+  handleChangePasswordError = err => {
+    const isWrongCurrentPassword = err.status === 401;
+
+    if (isWrongCurrentPassword) {
+      throw new SubmissionError({
+        currentPassword: this.translate('wrongPassword')
+      });
+    }
+  };
+
+  validateForm = values => {
+    const errors = {};
+    const enterValueError = this.translate('enterValue');
+
+    if (!values.currentPassword) {
+      errors.currentPassword = enterValueError;
+    }
+
+    if (!values.newPassword) {
+      errors.newPassword = enterValueError;
+    }
+
+    if (!values.confirmPassword) {
+      errors.confirmPassword = enterValueError;
+    }
+
+    const isFormValid = (
+      values.newPassword &&
+      values.confirmPassword &&
+      values.newPassword !== values.confirmPassword
+    );
+
+    if (isFormValid) {
+      const confirmPasswordMatchError = this.translate('confirmPasswordMatchError');
+
+      errors.confirmPassword = confirmPasswordMatchError;
+    }
+
+    return errors;
+  };
+
+  translate = id => {
+    const { stripes } = this.props;
+    const { intl: { formatMessage } } = stripes;
+
+    return formatMessage({ id: `${this.translateNamespace}.${id}` });
+  };
+
+  createCalloutRef = ref => { this.callout = ref; };
+
+  render() {
+    const { passwordMasked } = this.state;
+    const { label } = this.props;
+    const passwordType = passwordMasked ? 'password' : 'text';
+    const passwordToggleLabelId = `${this.translateNamespace}.${passwordMasked ? 'show' : 'hide'}Password`;
+
+    return (
+      <div style={this.styles.changePasswordFormWrap}>
+        <ChangePasswordForm
+          title={label}
+          validate={this.validateForm}
+          saveButtonText={this.translate('save')}
+          onSubmit={this.onChangePasswordFormSubmit}
+        >
+          <Row>
+            <Col xs={6}>
+              <Field
+                component={TextField}
+                type={passwordType}
+                id="current-password"
+                name="currentPassword"
+                label={this.translate('currentPassword')}
+                autoFocus
+              />
+            </Col>
+          </Row>
+          <Row>
+            <Col xs={6}>
+              <Field
+                component={TextField}
+                type={passwordType}
+                id="new-password"
+                name="newPassword"
+                label={this.translate('newPassword')}
+              />
+            </Col>
+          </Row>
+          <Row>
+            <Col xs={6}>
+              <Field
+                component={TextField}
+                type={passwordType}
+                id="confirm-password"
+                name="confirmPassword"
+                label={this.translate('confirmPassword')}
+              />
+            </Col>
+            <Col>
+              <div style={this.styles.toggleMaskButtonWrap}>
```

Comment:
What does the "Wrap" at the end of this style-name mean?

Reply:
It is just a wrapper on the button - though I need it here in order to attach styles to it.

Reply:
Then perhaps "Wrapper" would be better -- I wasn't sure whether to read this as referring to word-wrapping at the ends of lines.